processing/data_collection/gazette/spiders/sp_bauru.py

```diff
@@ -0,0 +1,55 @@
+import dateparser
+import itertools
+
+from itertools import product
+from datetime import datetime
+from scrapy import Request
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+class SpBauruSpider(BaseGazetteSpider):
+    MUNICIPALITY_ID = "3506003"
+
+    MONTHS = [str(x).zfill(2) for x in range(1, 13)]
+    YEARS = ["2015", "2016", "2017", "2018"]
+    MONTHS_YEARS = itertools.product(MONTHS, YEARS)
+
+    BASE_URL = "http://www.bauru.sp.gov.br{}"
+    PAGE_URL = "http://www.bauru.sp.gov.br/juridico/diariooficial.aspx?a={}&m={}"
+    DATE_CSS = "b::text"
+    PDF_HREF_CSS = "a::attr(href)"
+    GAZETTE_CSS = "main div.container ul ul ul li"
+
+    allowed_domains = ["bauru.sp.gov.br"]
+    name = "sp_bauru"
+
+    def start_requests(self):
+        for month, year in self.MONTHS_YEARS:
```

Comment:
Looks like this will also request URLs for future months in the current year (e.g. December 2018). Shouldn't we avoid that?